Xamarin.Forms.Core/Internals/PropertyPropagationExtensions.cs

```diff
@@ -21,5 +21,17 @@ public static void PropagatePropertyChanged(string propertyName, Element element
 					view.PropagatePropertyChanged(propertyName);
 			}
 		}
+
+		internal static void PropagatePropertyChanged(string propertyName, Element element)
+		{
+			if (propertyName == null || propertyName == VisualElement.FlowDirectionProperty.PropertyName)
+				Element.SetFlowDirectionFromParent(element);
+
+			if (propertyName == null || propertyName == VisualElement.VisualProperty.PropertyName)
+				Element.SetVisualfromParent(element);
```

Comment:
Shouldn't these two properties be in the `VisualElement` type with the `IPropertyPropagationController` interface added? That way, this method can just call it, and each type is responsible for its own properties.